File: code_samples/ruby/utilities_math_operations.rb
# Utilities: Math Operations
# AI/ML Training Sample

class MathOperations
  attr_accessor :data
  
  def initialize
    @data = ""
  end
  
  def process(input)
    @data = input
  end
  
  def validate
    !@data.empty?
  end
  
  def to_s
    "Data: #{@data}"
  end
end

# Example usage
instance = MathOperations.new
instance.process("example")
puts instance
puts "Valid: #{instance.validate}"

File: code_samples/r/utilities_math_operations.r
# Utilities: Math Operations
# AI/ML Training Sample

MathOperations <- setRefClass(
    "MathOperations",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- MathOperations$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")
